architecture-building-systems/CityEnergyAnalyst
# 1971 check input files

This pull request addresses 1971 by checking the input files for each script before running it via `cea.api`.

The logic used depends on `scripts.yml` having containing a list of input-files for each script as determined by their locator. This is... maybe not the best way to do it, but works quickly with what we currently have. A better solution would be to move scripts configuration into classes similar to plots, but this is out of scope for the MuSES project - we'd have to revisit each script and we really just don't have time for this.

The information for `scripts.yml` input-files was taken from `schemas.yml` which is currently being updated using @Jack-Hawthorne's work. I expect newer results tomorrow.

To test:
- run a script from the Dashboard that you know works (operation-costs runs quickly)
- delete an input file for that script
- run the script again

expected output is something like this:

```
City Energy Analyst version 2.16
Running `cea operation-costs` with the following parameters:
- general:scenario = C:\Users\daren-thomas\Downloads\reference-case-cooling\baseline
  (default: C:\reference-case-open\baseline)

---------------------------
ERROR: Missing input files:
- C:\Users\daren-thomas\Downloads\reference-case-cooling\baseline\outputs\data\demand\Total_demand.csv
  - try running demand
```
The output includes a hint on what tool to run to fix the error.
